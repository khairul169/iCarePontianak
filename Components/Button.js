import React from "react";
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import PropTypes from "prop-types";

const Button = ({
  title,
  backgroundColor,
  color,
  height,
  border,
  style,
  onPress
}) => {
  const containerStyle = [
    styles.container,
    backgroundColor && { backgroundColor },
    !border && {
      borderWidth: 0
    },
    style
  ];

  return (
    <View style={containerStyle}>
      <TouchableOpacity
        style={[styles.button, height && { height }]}
        onPress={onPress}
      >
        <Text style={[styles.buttonTitle, color && { color }]}>
          {title && title.toUpperCase()}
        </Text>
      </TouchableOpacity>
    </View>
  );
};

Button.propTypes = {
  title: PropTypes.string,
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  height: PropTypes.number,
  border: PropTypes.bool,
  style: PropTypes.any,
  onPress: PropTypes.func
};

Button.defaultProps = {
  border: true
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    borderColor: "#ccc",
    borderWidth: 1,
    borderRadius: 2
  },
  button: {
    alignItems: "center",
    justifyContent: "center",
    height: 40
  },
  buttonTitle: {
    color: "#626262",
    fontSize: 14,
    fontWeight: "bold"
  }
});

export default Button;
