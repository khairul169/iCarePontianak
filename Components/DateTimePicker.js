import React, { Component } from "react";
import DTPicker from "@react-native-community/datetimepicker";
import moment from "moment";
import PropTypes from "prop-types";

class DateTimePicker extends Component {
  static propTypes = {
    onValueChange: PropTypes.func
  };

  state = {
    date: new Date(),
    time: new Date(),
    mode: null
  };

  _dateChanged = (event, value) => {
    const { mode } = this.state;
    const { onValueChange } = this.props;

    let date = this.state.date;
    let time = this.state.time;

    switch (mode) {
      case "date":
        date = value || date;
        break;

      case "time":
        time = value || time;
        break;
    }

    this.setState({
      date,
      time,
      mode: mode === "date" ? "time" : null
    });

    if (mode === "time") {
      const dateTime =
        moment(date).format("YYYY-MM-DD") +
        " " +
        moment(time).format("HH:mm:ss");

      onValueChange && onValueChange(dateTime);
    }
  };

  show() {
    this.setState({
      mode: "date"
    });
  }

  hide() {
    this.setState({
      mode: null
    });
  }

  render() {
    const { mode, date, time } = this.state;

    return mode ? (
      <DTPicker
        value={mode === "date" ? date : time}
        mode={mode}
        is24Hour={true}
        onChange={this._dateChanged}
      />
    ) : null;
  }
}

export default DateTimePicker;
