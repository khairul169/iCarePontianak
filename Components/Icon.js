import React from "react";
import PropTypes from "prop-types";

// Icons
import Ionicons from "react-native-vector-icons/Ionicons";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";
import FontAwesome5 from "react-native-vector-icons/FontAwesome5";

const Icon = props => {
  let Icons = MaterialCommunityIcons;

  switch (props.type) {
    case "Ionicons":
      Icons = Ionicons;
      break;
    case "FontAwesome5":
      Icons = FontAwesome5;
      break;
    default:
      break;
  }

  return (
    <Icons
      name={props.name}
      size={props.size}
      color={props.color}
      style={props.style}
    />
  );
};

Icon.propTypes = {
  type: PropTypes.string,
  size: PropTypes.number,
  color: PropTypes.string,
  name: PropTypes.string.isRequired,
  style: PropTypes.object
};

Icon.defaultProps = {
  type: "MaterialCommunityIcons",
  size: 24,
  color: "#333"
};

export default Icon;
