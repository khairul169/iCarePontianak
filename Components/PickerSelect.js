import React from "react";
import { View, StyleSheet, Picker } from "react-native";
import PropTypes from "prop-types";

const PickerSelect = ({ items, value, onValueChange }) => {
  return (
    <View style={styles.container}>
      <Picker
        selectedValue={value}
        onValueChange={onValueChange}
        style={styles.picker}
      >
        {items &&
          items.map((item, index) => (
            <Picker.Item key={index} label={item.label} value={item.value} />
          ))}
      </Picker>
    </View>
  );
};

Picker.propTypes = {
  items: PropTypes.array,
  value: PropTypes.any,
  onValueChange: PropTypes.func
};

const styles = StyleSheet.create({
  container: {
    borderColor: "#ddd",
    borderWidth: 1,
    borderRadius: 3,
    paddingLeft: 8
  },
  picker: {
    height: 40
  }
});

export default PickerSelect;
