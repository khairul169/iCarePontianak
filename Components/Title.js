import React from "react";
import { Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const Title = ({
  style,
  children,
  fontSize,
  marginTop,
  marginBottom,
  marginX,
  color
}) => {
  const textStyle = [
    styles.title,
    {
      fontSize,
      marginTop,
      marginBottom,
      marginHorizontal: marginX,
      color
    },
    style
  ];
  return <Text style={textStyle}>{children}</Text>;
};

Title.propTypes = {
  style: PropTypes.any,
  fontSize: PropTypes.number,
  marginTop: PropTypes.number,
  marginBottom: PropTypes.number,
  marginX: PropTypes.number,
  color: PropTypes.string
};

Title.defaultProps = {
  fontSize: 16,
  marginTop: 0,
  marginBottom: 16,
  marginX: 0,
  color: "#37474F"
};

const styles = StyleSheet.create({
  title: {
    fontSize: 16,
    color: "#525252"
  }
});

export default Title;
