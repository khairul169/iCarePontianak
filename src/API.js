import axios from "axios";
import { APP_API } from "react-native-dotenv";

const RELEASE = true;
const API_URL = RELEASE ? APP_API : "http://192.168.43.48/icare/public/";

const auth = token => {
  return {
    headers: token
      ? {
          Authorization: `Bearer ${token}`
        }
      : undefined
  };
};

class API {
  static async get(url, token) {
    return axios
      .get(API_URL + url, auth(token))
      .then(result => {
        console.log(result.data);
        return result.data;
      })
      .catch(error => {
        console.log(error.message);
      });
  }

  static async post(url, body, token) {
    return axios
      .post(API_URL + url, body, auth(token))
      .then(result => {
        console.log(result.data);
        return result.data;
      })
      .catch(error => {
        console.log(error.message);
      });
  }

  static async patch(url, body, token) {
    return axios
      .patch(API_URL + url, body, auth(token))
      .then(result => {
        console.log(result.data);
        return result.data;
      })
      .catch(error => {
        console.log(error.message);
      });
  }
}

export default API;
