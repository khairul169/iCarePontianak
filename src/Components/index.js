import Card from "./Card";
import Icon from "./Icon";

export { Card, Icon };
