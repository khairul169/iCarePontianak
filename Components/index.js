import BottomSheet from "./BottomSheet";
import Button from "./Button";
import Card from "./Card";
import DateTimePicker from './DateTimePicker'
import Header from "./Header";
import HomeHeader from "./HomeHeader";
import Icon from "./Icon";
import ItemLayanan from "./ItemLayanan";
import MapLayout from "./MapLayout";
import PickerSelect from "./PickerSelect";
import SearchBar from "./SearchBar";
import StatusBar from "./StatusBar";
import TextEdit from "./TextEdit";
import Title from "./Title";

export {
  BottomSheet,
  Button,
  Card,
  DateTimePicker,
  Header,
  HomeHeader,
  Icon,
  ItemLayanan,
  MapLayout,
  PickerSelect,
  SearchBar,
  StatusBar,
  TextEdit,
  Title
};
