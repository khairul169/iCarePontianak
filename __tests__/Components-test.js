import React from "react";
import renderer from "react-test-renderer";

import { Card, Icon } from "../src/Components";

describe("<Card />", () => {
  const defaultProps = {
    flex: 1,
    margin: 10,
    marginX: 2,
    marginY: 6
  };
  const wrapper = renderer.create(<Card {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<Icon />", () => {
  const defaultProps = {
    type: "Ionicons",
    name: "ios-home",
    size: 24,
    color: "#333"
  };
  const wrapper = renderer.create(<Icon {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
