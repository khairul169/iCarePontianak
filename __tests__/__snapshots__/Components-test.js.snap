// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "borderRadius": 5,
        "elevation": 5,
        "flex": 1,
        "margin": 10,
        "marginBottom": undefined,
        "marginHorizontal": 2,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": 6,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Icon /> render 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#333",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ï‘ˆ
</Text>
`;
